src/plugins_3rdparty/primer3/src/Primer3Task.cpp

```diff
@@ -668,52 +654,28 @@ QList<Task*> Primer3ToAnnotationsTask::onSubTaskFinished(Task* subTask) {
-                int firstExonIdx = exonRange.minValue;
-                int lastExonIdx = exonRange.maxValue;
-                if (firstExonIdx > regions.size()) {
-                    setError(tr("The first exon from the selected range [%1,%2] is larger the number of exons (%3)."
-                                " Please set correct exon range.")
-                                 .arg(firstExonIdx)
-                                 .arg(lastExonIdx)
-                                 .arg(regions.size()));
-                    return res;
-                }
-
-                if (lastExonIdx > regions.size()) {
-                    setError(tr("The the selected exon range [%1,%2] is larger the number of exons (%3)."
-                                " Please set correct exon range.")
-                                 .arg(firstExonIdx)
-                                 .arg(lastExonIdx)
-                                 .arg(regions.size()));
-                    return res;
-                }
-
-                regions = regions.mid(firstExonIdx - 1, lastExonIdx - firstExonIdx + 1);
-                int totalLen = 0;
-                for (const U2Region& r : regions) {
-                    totalLen += r.length;
-                }
-                settings->setIncludedRegion(regions.first().startPos + settings->getFirstBaseIndex(), totalLen);
+        findExonTaskIsfinished();
+        CHECK_OP(stateInfo, res);
+
+        primer3Task = new Primer3Task(settings);
+        res.append(primer3Task);
+    } else if (subTask == primer3Task && checkComplementTask == nullptr && settings->getCheckComplementSettings().enabled) {
+        checkComplementTask = new CheckComplementTask(settings->getCheckComplementSettings(), primer3Task->getBestPairs(), seqObj);
+        res.append(checkComplementTask);
+    } else if (subTask == primer3Task || subTask == checkComplementTask) {
+        CHECK_EXT(!annotationTableObject.isNull(), setError(tr("Object with annotations was removed")), res);
+        SAFE_POINT(primer3Task != nullptr, L10N::nullPointerError("Primer3Task"), res);
+
+        auto resultAnnotations = getResultAnnotations();
+        if (resultAnnotations.isEmpty()) {
+            if (primer3Task->getBestPairs().isEmpty()) {
+                stateInfo.addWarning(tr("No primers has been found due to the parameters you've set up"));
+            } else {
+                stateInfo.addWarning(tr("All found primers has been filtered due to the \"Check complement\" parameters"));
             }
-            settings->setExonRegions(regions);
-            // reset target and excluded regions regions
-            QList<U2Region> emptyList;
-            settings->setExcludedRegion(emptyList);
-            settings->setTarget(emptyList);
         }
 
-        searchTask = new Primer3SWTask(settings);
-        res.append(searchTask);
+        res.append(new CreateAnnotationsTask(annotationTableObject, resultAnnotations));
```

Comment:
Do not create the task when resultAnnotations is empty.

Reply:
OK